### ISSUE TITLE
map will play normally in skirmish,but game crashes when is used in multiplayer server

### ISSUE THREAD

**Issue by KOYK** [OpenRA/OpenRA#12940]
The map is unable to start on a multiplayer server,and by clicking at the "options" in the game lobby the game crashes. the map works fine in skirmish.
I post this as an issue because i find weird that the skirmish can use the map and the multiplayer servers can't.

map: 
http://resource.openra.net/maps/19854

exception log: 
Red Alert Mod at Version release-20161019
Date: 2017-03-09 09:35:38Z
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.Collections.Generic.KeyNotFoundException`: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at OpenRA.Mods.Common.Widgets.Logic.LobbyLogic.<LobbyLogic>c__AnonStoreyA.<>m__0(Global gs)
   at OpenRA.Widgets.CachedTransform`2.Update(T input)
   at OpenRA.Mods.Common.Widgets.Logic.LobbyLogic.<LobbyLogic>c__AnonStoreyA.<>m__5()
   at OpenRA.Mods.Common.Widgets.ButtonWidget.Draw()
   at OpenRA.Mods.Common.Widgets.DropDownButtonWidget.Draw()
   at OpenRA.Widgets.Widget.DrawOuter()
   at OpenRA.Widgets.Widget.DrawOuter()
   at OpenRA.Widgets.Widget.DrawOuter()
   at OpenRA.Widgets.Widget.DrawOuter()
   at OpenRA.Widgets.Widget.DrawOuter()
   at OpenRA.Widgets.Widget.DrawOuter()
   at OpenRA.Game.RenderTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Program.Run(String[] args)
   at OpenRA.Program.Main(String[] args)

**Comment by KOYK** [comment]
ok i found what cause the problem,it was a duplicated code.by adding the lines
```
World:
	AllyBuildRadiusLocked: True
	AllyBuildRadiusEnabled: True
```
to a secondary .yaml rules file.

but the question remains,why the skirmish mode din't get effected?

**Comment by rob-v** [comment]
I worked today on this issue https://github.com/OpenRA/OpenRA/issues/12881. I fixed it, sent pull request, but it needs review as I am not sure about all my assumptions. 
I wanted to verify it with your map, but it works fine, it is the corrected one I assume. It doesn't work in Multiplayer, because it fails only when sending such duplicated GlobalSettings.LobbyOptions to another OpenRA client.

**Comment by KOYK** [comment]
thank you for your work!!

**Comment by abcdefg30** [comment]
Fixed by #12976.
